### PR TITLE
cicd: testing out integration test

### PR DESCRIPTION
**Pull request by bernadinm**
Testing out Integration

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -35,5 +35,18 @@ pipeline {
...
         }
+        stage('Integration Test') {
+            agent { label "dcos-terraform-cicd" }
+            steps {
+                withCredentials([
+                   [$class: 'AmazonWebServicesCredentialsBinding', credentialsId: 'dcos-terraform-ci-aws'],
+                   azureServicePrincipal('dcos-terraform-ci-azure'),
+                   [$class: 'FileBinding', credentialsId: 'dcos-terraform-ci-gcp', variable: 'GOOGLE_APPLICATION_CREDENTIALS']
+                   ]) {
+                         sh 'printenv'
```
Comment (fatz) [inline Jenkinsfile]: please remove. There is no need showing the environment.

File: ci-deploy.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -o nounset -o errexit
+set -x
+
+TMP_DCOS_TERRAFORM=$(mktemp -d)
+git clone https://github.com/bernadinm/dcos-terraform.git $TMP_DCOS_TERRAFORM
```
Comment (fatz) [inline ci-deploy.sh]: please use either the convenience wrapper or the examples repo. Later would probably the best bet.

File: ci-deploy.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+cd $TMP_DCOS_TERRAFORM
+make init
+rm -fr $TMP_DCOS_TERRAFORM/$(grep $(echo $GIT_URL | cut -d'/' -f 5 | cut -d'.' -f1) $TMP_DCOS_TERRAFORM/.gitmodules -B1 | grep path | cut -d' ' -f3)
+cp -fr $WORKSPACE $TMP_DCOS_TERRAFORM/$(grep $(echo $GIT_URL | cut -d'/' -f 5 | cut -d'.' -f1) $TMP_DCOS_TERRAFORM/.gitmodules -B1 | grep path | cut -d' ' -f3)
+PROVIDER="$(git diff master --name-only | grep -E '^modules|^example' | cut -d'/' -f2 | grep -vE 'null|template|localfile|.git*' | sort | uniq | xargs)" make tenv | tee -a env_to_run.log
+for i in $(grep -E '_output.*examples' env_to_run.log | sort | uniq | cut -d' ' -f2); do
+  cd $i;
+  eval $(ssh-agent); rm ssh-key.pub; ssh-keygen -t rsa -b 4096 -f ssh-key -P ''; ssh-add ssh-key
+  terraform apply -auto-approve;
+  terraform destroy -auto-approve;
```
Comment (fatz) [inline ci-deploy.sh]: this will only destroy when the job was sucessfull. Post build actions is what you want to do

File: ci-deploy.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+TMP_DCOS_TERRAFORM=$(mktemp -d)
+git clone https://github.com/bernadinm/dcos-terraform.git $TMP_DCOS_TERRAFORM
+cd $TMP_DCOS_TERRAFORM
+make init
+rm -fr $TMP_DCOS_TERRAFORM/$(grep $(echo $GIT_URL | cut -d'/' -f 5 | cut -d'.' -f1) $TMP_DCOS_TERRAFORM/.gitmodules -B1 | grep path | cut -d' ' -f3)
+cp -fr $WORKSPACE $TMP_DCOS_TERRAFORM/$(grep $(echo $GIT_URL | cut -d'/' -f 5 | cut -d'.' -f1) $TMP_DCOS_TERRAFORM/.gitmodules -B1 | grep path | cut -d' ' -f3)
+PROVIDER="$(git diff master --name-only | grep -E '^modules|^example' | cut -d'/' -f2 | grep -vE 'null|template|localfile|.git*' | sort | uniq | xargs)" make tenv | tee -a env_to_run.log
+for i in $(grep -E '_output.*examples' env_to_run.log | sort | uniq | cut -d' ' -f2); do
+  cd $i;
+  eval $(ssh-agent); rm ssh-key.pub; ssh-keygen -t rsa -b 4096 -f ssh-key -P ''; ssh-add ssh-key
```
Comment (fatz) [inline ci-deploy.sh]: please use https://jenkins.io/doc/pipeline/steps/ssh-agent/ instead

**Review by fatz — CHANGES_REQUESTED** [review]
Hey this only works in ideal scenarios. If there is something wrong the cluster will not be cleaned up. Also a real test scenario is missing. The cluster could still be broken.

Furthermore the .tfstate should be saved somewhere to debug errors.

Please have a look into the not public "integrationtest" example in Jenkins.

File: ci-deploy.sh (hunk 2 above)
Comment (bernadinm) [reply ci-deploy.sh]: Yes, the convenience wrapper is used. It's sourcing the examples here for each cloud https://github.com/bernadinm/dcos-terraform/tree/master/examples

File: Jenkinsfile (hunk 1 above)
Comment (bernadinm) [reply Jenkinsfile]: yep! Sounds good!

File: ci-deploy.sh (hunk 3 above)
Comment (bernadinm) [reply ci-deploy.sh]: Thanks @fatz. Its done.

**Comment by sebbrandt87** [comment]
this can now be deleted

**Pull request closed without merging**